components/FlashcardForm.js

```diff
@@ -0,0 +1,57 @@
+import { collections } from "@/lib/db/collections";
+import styled from "styled-components";
+
+const StyledForm = styled.form`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  gap: 4px;
+`;
+
+export default function FlashcardForm({ onSubmit }) {
+  function handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+
+    onSubmit(data);
+  }
```

Comment:
very nice separation of the Form functionality (getting the data from the inputs) and the flash card creation (everthing inside the onSubmit function).